docs/concepts/workloads/vm.md

```diff
@@ -84,15 +84,45 @@ It is possible to update parts of an existing `VirtualMachine` resource. Some fi
+Some of a VM's hardware resources are derived from the policies defined by your infrastructure administrator, others may be influenced directly by a user.
 
 ### CPU and Memory
+CPU and memory of a VM are derived form the `VirtualMachineClass` that is used to create the VM. Specifically, the `memoryMB` and `numCPUs` properties in the `configSpec` field of the `VirtualMachineClass` resource dictate the exact number of virtual CPUs, and the virtual memory (in MegaBytes) that the VM will be created with.
+
+Additionally, your administrator might also define certain policies in your VM class in form of resource [reservations](https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.resmgmt.doc/GUID-8B88D3D8-E9D9-4C05-A065-B3DE1FFFB401.html) or [limits](https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.resmgmt.doc/GUID-117972E3-F5D3-4641-9EAC-F9DD2B0761C3.html). These are vSphere constructs which are used to reserve, or set a ceiling on resources consumed by a VM. Users can view these policies by inspecting the `ConfigSpec.cpuAllocation` and `ConfigSpec.memoryAllocation` fields of the VM Class for CPU and memory reservation/limits information respectively.
+
+For an example, consider the following VM Class:
+```
+apiVersion: vmoperator.vmware.com/v1alpha1
+kind: VirtualMachineClass
+metadata:
+  name: my-vm-class
+spec:
+  configSpec:
+    _typeName: VirtualMachineConfigSpec
+    cpuAllocation:
+      _typeName: ResourceAllocationInfo
+      limit: 400
+      reservation: 200
+    memoryAllocation:
+      _typeName: ResourceAllocationInfo
+      limit: 2048
+      reservation: 1024
+  hardware:
+    cpus: 2
+    memory: 4Gi
+```
+A VM created with this VM class will have 2 vCPUs, 4 GiB of memory. Additionally, the VM will be created with a guaranteed 200MHz of CPU bandwidth and 1024 MB of memory. The CPU bandwidth of the VM will never be allowed to exceed 400MHz and the memory will not exceed 4Gi.
+
+!!! note "Units for CPU reservations/limits are in MHz"
+
+Please note that the units for CPU are different in hardware and reservations/limits.  Virtual hardware is specified in units of vCPUs, whereas CPU reservations/limits are in MHz.  Memory can be specified in MiB, GiB, whereas memory reservations/limits are always in MB.
 
 ### Networking
+Users can specify a list of network interfaces for their VM in the `spec.NetworkInterfaces` field. Each interface must specify the network type (one of `nsx-t`, or `vsphere-distributed` depending on the network provider being used). Users can further customize each interface with optional fields such as the Ethernet card, network name etc. For convenience, even if a VM does not specify any network interface, VM operator creates one connected to the default network available in the namespace. The IP address of the network interface is handled by the available network provider (configured by the administrator).
 
```

Comment:
Do we want to callout `spec.NetworkInterfaces` can be optional in the first sentence? 
Also maybe add a code snippet will help here?
```
apiVersion: vmoperator.vmware.com/v1alpha1
kind: VirtualMachine
metadata:
 name: ubuntu-vm
 namespace: test-ns
spec:
 networkInterfaces:
 - networkName: "primary"
  networkType: vsphere-distributed
  ```

Reply:
I changed the text to indicate that the network interface is optional.  I also modified the last line to make it more clear that the default network interface is created only when nothing is specified.  I don't think adding an example here is required (because we are not adding an example for any other fields such as volumes).